openamp: resource table: fixes resource table to work with ram_console section

mausys opened the pull request:
in commit 1cd37f21f3a the global ram console buffer was replaced with a pointer. This didn't work with the OpenAMP resource table anymore (#75656). This commit removes also the unnecessary console buffer, if a ram console device tree node is configured.
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/75656

carlocaione commented:
Hi @mausys can you make sure that the CI is all green?

mausys commented:
This change is also included in #79605

mmahadevan108 commented:
cc @alxlastur @Zhiqiang-Hou
Can you also please help review this PR

mmahadevan108 commented on lib/open-amp/resource_table.c:
```diff
@@ -29,7 +29,18 @@
+#else
+#error "Lack of chosen property zephyr,ram_console!"
+#endif
+
```
Is'nt it sufficient to just use the extern to get the already defined `ram_console_buf`. This has the `RAM_CONOLE_BUF_ATTR` defined for it which should place it in the appropriate RAM section.

mausys replied:
The ram_console_buf in ram_console.c is only used when CONFIG_RAM_CONSOLE_BUFFER_SECTION is disabled. That's why I changed it in ram_console.c to not wasting memory.

Zhiqiang-Hou commented on drivers/console/ram_console.c:
```diff
@@ -29,7 +29,10 @@
+#ifndef CONFIG_RAM_CONSOLE_BUFFER_SECTION
 char ram_console_buf[CONFIG_RAM_CONSOLE_BUFFER_SIZE] RAM_CONSOLE_BUF_ATTR;
+#endif
+
```
revert this and directly use `extern char ram_console_buf[];` in lib/open-amp/resource_table.c.

mausys committed:
openamp: resource table: fixes resource table for ram_console section

in commit 1cd37f21f3a the global ram console buffer was replaced
with a pointer. This didn't work with the OpenAMP resource table
anymore (#75656).

Signed-off-by: Simon Maurer <mail@maurer.systems>

mmahadevan108 commented:
@carlocaione, can you help review this again.

mmahadevan108 commented:
@carlocaione , can you help take a look. This bug fix can be included in 4.0 release
